### PR TITLE
feat(mentions) mention teams!

### PR DESCRIPTION
**Pull request by MaxBittker**
this commit adds the note input, group note serializer, and group note endpoint changes to allow you to mention teams with `#` .

Introduces the "Actor" class, which is a union of users and teams.

### CODE REVIEW

**Comment by getsentry-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">You should update CHANGES due to the size of this PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

File: src/sentry/api/endpoints/group_notes.py
Comment (macqueen) [inline src/sentry/api/endpoints/group_notes.py]: did you intentionally move this? because now you're checking to see if the activity exists with a different data object than you're creating it with below

File: src/sentry/api/endpoints/group_notes.py
Comment (macqueen) [inline src/sentry/api/endpoints/group_notes.py]: you're gonna want to use `select_related` here to avoid running a query when accessing the user for every member. I think actually since `member_set` returns a queryset, you should be able to just chain  `team.member_set.select_related('user')`

File: src/sentry/api/serializers/rest_framework/group_notes.py
Comment (macqueen) [inline src/sentry/api/serializers/rest_framework/group_notes.py]: i think some comments about what these two functions do/how they are different could be helpful

File: src/sentry/api/serializers/rest_framework/group_notes.py
Comment (macqueen) [inline src/sentry/api/serializers/rest_framework/group_notes.py]: i know you didn't name this variable, but this should probably be `user_ids` since the current name implies `member.id`

**Review by macqueen — COMMENTED** [review]
generally seems 👍  to me, aside from a few comments. i'll defer to matt on the actor class.

File: src/sentry/api/serializers/rest_framework/group_notes.py
Comment (MaxBittker) [reply src/sentry/api/serializers/rest_framework/group_notes.py]: good point

File: src/sentry/static/sentry/app/components/activity/noteInput.jsx
Comment (billyvg) [inline src/sentry/static/sentry/app/components/activity/noteInput.jsx]: This and `mentionableUsers` need to go into state otherwise you'll be calling these methods on every render (and that happens on every keystroke I'm guessing).

Comment (MaxBittker) [reply src/sentry/static/sentry/app/components/activity/noteInput.jsx]: 👍

Comment (MaxBittker) [reply src/sentry/static/sentry/app/components/activity/noteInput.jsx]: good catch, that was slightly hand wavy of me because i thought about it when i moved it but didn't consider that it would be each keystroke and could be a large number of members being mapped over

**Comment by billyvg** [comment]
JS lgtm

**Review by mattrobenolt — CHANGES_REQUESTED** [review]

File: src/sentry/api/endpoints/group_notes.py
Comment (mattrobenolt) [reply src/sentry/api/endpoints/group_notes.py]: I guess one other minor nit: `actor_mentions`, not `actorMentions`. We're writing 🐍 @MaxBittker.

File: src/sentry/api/serializers/rest_framework/group_notes.py
Comment (MaxBittker) [reply src/sentry/api/serializers/rest_framework/group_notes.py]: my opinion is think this is not worth a really small multiplier on the performance here. if you feel strongly i'll do it though

File: src/sentry/models/groupsubscription.py
Comment (MaxBittker) [reply src/sentry/models/groupsubscription.py]: gonna punt on this, it's not trivial

File: src/sentry/api/serializers/rest_framework/group_notes.py
Comment (mattrobenolt) [reply src/sentry/api/serializers/rest_framework/group_notes.py]: I don't think I feel strongly enough to block on it, I just don't think it's better. :)

File: src/sentry/api/serializers/rest_framework/group_notes.py
Comment (MaxBittker) [reply src/sentry/api/serializers/rest_framework/group_notes.py]: I was worried that there were teams or users ids that had been merged somehow, but this is probably safe to drop

File: src/sentry/api/endpoints/group_notes.py
Comment (MaxBittker) [reply src/sentry/api/endpoints/group_notes.py]: will do in followup PR

File: src/sentry/api/serializers/rest_framework/group_notes.py
Comment (mattrobenolt) [inline src/sentry/api/serializers/rest_framework/group_notes.py]: ProTip: `{user.id for user in users}`

File: src/sentry/api/endpoints/group_notes.py
Comment (mattrobenolt) [inline src/sentry/api/endpoints/group_notes.py]: Do you need to do this anymore? Couldn't we just `exclude` them from the queryset above?

```python
mentioned_team_users = User.object.filter(...).exclude(id__in=subscribed_user_ids)
```

This whole thing feels a bit clunky imo. Since `subscribed_user_ids` itself is just derived from `actor_mentions` It seems like we could drop the variable entirely and just use:

```python
.exclude(id__in={u.id for u in actor_mentions['users']})
```

And avoid the extra bookkeeping.

**Review by mattrobenolt — APPROVED** [review]
Other than my last nit about `subscribed_user_ids`, looks good enough to ship. :cookie:

**Pull request merged**
